src/components/NFTViewOptionList.vue

```diff
@@ -183,50 +184,41 @@ export default {
           return this.$t('nft_details_page_button_view_unknown');
       }
     },
-    getFilenameFromURL(url) {
-      const qsStr = url.split('?').pop();
-      const qs = querystring.parse(qsStr);
-      return qs?.name || '';
-    },
-    getDownloadFilenameFromURL(url) {
-      return `${this.getFilenameFromURL(url) || 'content'}`;
-    },
+    getFilenameFromURL,
+    getDownloadFilenameFromURL,
     handleClickViewContent() {
       this.$emit('view-content');
     },
+    async passCORSPolicy(url) {
+      try {
+        await this.$axios.head(url);
```

Comment:
Not all GET resources support HEAD though, safest bet is try a GET and ignore the body after we get the 200 status

from chatGPT

```
axios.get('https://example.com/api/endpoint', { responseType: 'stream' })
  .then(response => {
    if (response.status === 200) {
      console.log('GET request succeeded');
      // You can choose to abort the request here if you don't need the response body
      response.data.destroy();
    } else {
      console.log('GET request failed');
    }
  })
  .catch(error => {
    console.log('An error occurred:', error);
  });
  ```

Reply:
Nice trick, thanks!!